sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredentialBuilder.java

```diff
@@ -272,44 +273,71 @@ private ArrayList<TokenCredential> getCredentialsChain() {
-        boolean useProductionCredentials = false;
-        boolean useDeveloperCredentials = false;
-        if (!CoreUtils.isNullOrEmpty(selectedCredentials)) {
-            selectedCredentials = selectedCredentials.trim();
-            if ("prod".equalsIgnoreCase(selectedCredentials)) {
-                useProductionCredentials = true;
-            } else if ("dev".equalsIgnoreCase(selectedCredentials)) {
-                useDeveloperCredentials = true;
+
+        String selectedCredential = configuration.get("AZURE_TOKEN_CREDENTIALS");
+        ArrayList<TokenCredential> credentials = new ArrayList<>(8);
+
+        if (!CoreUtils.isNullOrEmpty(selectedCredential)) {
+            selectedCredential = selectedCredential.trim().toLowerCase(Locale.ROOT);
+
+            // Use a map to associate credential names to their adders
+            java.util.Map<String, Runnable> credentialMap = new java.util.HashMap<>();
+            credentialMap.put("prod", () -> addProdCredentials(credentials));
+            credentialMap.put("dev", () -> addDevCredentials(credentials));
+            credentialMap.put("environmentcredential",
+                () -> credentials.add(new EnvironmentCredential(identityClientOptions.clone())));
+            credentialMap.put("workloadidentitycredential", () -> credentials.add(getWorkloadIdentityCredential()));
+            credentialMap.put("managedidentitycredential",
+                () -> credentials.add(new ManagedIdentityCredential(managedIdentityClientId, managedIdentityResourceId,
+                    null, identityClientOptions.clone())));
+            credentialMap.put("intellijcredential",
+                () -> credentials.add(new IntelliJCredential(tenantId, identityClientOptions.clone())));
+            credentialMap.put("azureclicredential",
+                () -> credentials.add(new AzureCliCredential(tenantId, identityClientOptions.clone())));
+            credentialMap.put("azurepowershellcredential",
+                () -> credentials.add(new AzurePowerShellCredential(tenantId, identityClientOptions.clone())));
+            credentialMap.put("azuredeveloperclicredential",
+                () -> credentials.add(new AzureDeveloperCliCredential(tenantId, identityClientOptions.clone())));
+            credentialMap.put("visualstudiocodecredential",
+                () -> credentials.add(new VisualStudioCodeCredential(tenantId, identityClientOptions.clone())));
+
+            Runnable adder = credentialMap.get(selectedCredential);
+            if (adder != null) {
+                adder.run();
+                return credentials;
             } else {
-                throw LOGGER.logExceptionAsError(new IllegalArgumentException(
-                    "Invalid value for AZURE_TOKEN_CREDENTIALS. Valid values are 'prod' or 'dev'."));
+                throw LOGGER
+                    .logExceptionAsError(new IllegalArgumentException("Invalid value for AZURE_TOKEN_CREDENTIALS: '"
```

Comment:
Add an aka.ms link that points to the new TSG section

Reply:
added.